rmw_fastrtps_dynamic_cpp/src/rmw_client.cpp

```diff
@@ -51,24 +54,30 @@ rmw_create_client(
   const rosidl_service_type_support_t * type_supports,
   const char * service_name, const rmw_qos_profile_t * qos_policies)
 {
-  if (!node) {
-    RMW_SET_ERROR_MSG("node handle is null");
+  RMW_CHECK_ARGUMENT_FOR_NULL(node, nullptr);
```

Comment:
why all the error handling changes were applied to `rmw_fastrtps_cpp` function and not here?

Reply:
Lack of time. For basic compliance with updated `rmw` API specs, changing return codes is enough. That's pretty much what happened to `rmw_fastrtps_dynamic_cpp`, and in https://github.com/ros2/rmw_cyclonedds/pull/247 and https://github.com/ros2/rmw_connext/pull/464. `rmw_fastrtps_cpp` got a bonus cleanup solely because it's the default implementation (and thus the one I care the most right now).

Reply:
I would like to keep `rmw_fastrtps_cpp` and `rmw_fastrtps_dynamic_cpp` close, but ok

Reply:
Yeah, I'm not super happy about it either, but I can't afford changing it right now. I promise I'll circle back later if time allows.